study-group/Toast-Implementation-Studies.md

```diff
@@ -90,9 +90,132 @@ The `restoreFocusOnClear` attribute restores page's focus once the toast is clea
 
 *TODO: Fill in study for this library*
 
-## ngx-toastr
+## [ngx-toastr](https://github.com/scttcper/ngx-toastr)
+ngx-toastr is a library from author [@scttcper](https://github.com/scttcper) to create a toast notification built for Angular.
 
-*TODO: Fill in study for this library*
+
+### Design & Sample Code
+Before using ngx-toastr, setup is required. First, add the CSS:
+
+```js
+// regular style toast
+@import '~ngx-toastr/toastr';
+
+// bootstrap style toast
+@import '~ngx-toastr/toastr-bs4-alert';
+```
+
+Next, add ToastrModule and BrowserAnimationsModule to app NgModule:
+
+```js
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { ToastrModule } from 'ngx-toastr';
+
+@NgModule({
+  imports: [
+    CommonModule,
+    BrowserAnimationsModule, // required animations module
+    ToastrModule.forRoot() // ToastrModule added
+  ],
+  ...
+})
+...
+```
+
+Once the modules are set up, this code will display a toast:
+
+```js
+import { ToastrService } from 'ngx-toastr';
+
+@Component({...})
+export class SomeComponent {
+  constructor(private toastr: ToastrService) {}
+
+  showSuccess() {
+    this.toastr.success('Hello title!', 'Hello message!', { timeOut: 3000 });
+  }
+}
+```
+
+To show the toast with an instance of ToastrService, use `toastr.success/error/warning/info/show()`, depending on the context and intent.
+The third parameter configures the toast with a variety of specialized options, for example:
+
+```js
+this.toastrService.error('Uh Oh...', '<b>Something is Wrong<b>', {
+  timeOut: 20000,
+  tapToDismiss: false,
+  enableHtml: true,
+  closeButton: true,
+  progressBar: true
+});
+```
+
+All these options can be configured globally like so:
+
+```js
+// root app NgModule
+imports: [
+  ToastrModule.forRoot({
+    timeOut: 10000,
+    positionClass: 'toast-bottom-right',
+    preventDuplicates: true,
+  }),
+],
+```
+
+These global options will be overriden by the individual toast configurations.
+For finer control, ngx-toaster allows both custom toast components and display in container, used like this:
```

Comment:
I'm confused by this example, `YourToastComponent` is never defined (even incompletely). Perhaps split this into 2, one for each feature. Also it's unclear to me what a custom toast component gets you. What does it override, why would I use one?

Reply:
That could definitely be more clear, the custom toast component overrides the view into a custom view that you create in an Angular Component. I'll add some example code and text to make that clearer.